### PR TITLE
Apply consistent Sharp library configuration

### LINKED ISSUE #20995: It seems that ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION doesn't change limitInputPixels
### Describe the Bug

When I am trying to download large image to the directus, I am receiving an error:
```json
{
  "errors": [
    {
      "message": "Input image exceeds pixel limit",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

It seems this error came from sharp library. I've tried to set this env value large enough, but it doesn't work
```bash
ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION=number:100000000000
```

My image has the next sizes: 16226 × 20665
So, pixels amount: 335310290
From Directus docs I know that `limitInputPixels=Math.pow(env['ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION'], 2)`. So my `limitInputPixels=1e+22` and this value much bigger than amount of pixels from my image.

Can you check it, pls?

### To Reproduce

Try to download a large image: here is my https://drive.google.com/file/d/1eqKWnfz2emyw0e-m26oh8BjUAdetvLzB/view?usp=sharing

### Directus Version

10.7.2

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by AndriyAntonenko**
## Scope

What's changed:

- Created the new util called `useSharp`, where all configurations are applied to the sharp instance and can be reused 
- Added test for the `useSharp` function
- Added me to the contributor's list
---

Fixes #20995

### CODE REVIEW

**Review by paescuj — COMMENTED** [review]
Thanks @AndriyAntonenko!
A few minor change requests, otherwise looking good to me 🚀

**Comment by AndriyAntonenko** [comment]
@paescuj Thanks for review. I've applied all requested changes

File: .changeset/silent-emus-rush.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/api': patch
+---
+
+Ensured `ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION` is also respected for extraction of metadata during image upload.
```
Comment (paescuj) [inline .changeset/silent-emus-rush.md]:
```suggestion
Ensured `ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION` is also respected for extraction of metadata during image upload
```

**Review by rijkvanzanten — APPROVED** [review]

**Comment by rijkvanzanten** [comment]
Thanks @AndriyAntonenko! ❤️ 🚀

**Comment by paescuj** [comment]
@rijkvanzanten Currently, this completely prevents updating images with a higher resolution (due to error in metadata extraction), but I guess we should just skip metadata extraction in that case 🤔

**Comment by rijkvanzanten** [comment]
> this completely prevents updating images with a higher resolution

What do you mean by this? Do you mean the "Replace file" functionality? Things are looking good on my end from what I can tell

**Comment by paescuj** [comment]
Sorry, I meant for images higher than `ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION` the upload will fail with `Input image exceeds pixel limit` AFAICS. But I guess we should still allow uploading such images, just skip metadata extraction and fail on download/transform.

**Pull request merged**